### PR TITLE
Adding a couple of new options to search.

### PR DESCRIPTION
**Pull request by alxdarksage**
Select the predicate for search terms and select string position match (prefix, postfix, infix). BRIDGE-3039.

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java [hunk 1]
```diff
@@ -124,51 +144,51 @@ QueryBuilder makeQuery(String prefix, String appId, AccountId accountId, Account
...
-                where.append("acct.phone.number=:number AND acct.phone.regionCode=:regionCode",
-                        "number", unguarded.getPhone().getNumber(),
-                        "regionCode", unguarded.getPhone().getRegionCode());
+                where.appendRequired("acct.phone.number=:number", "number", unguarded.getPhone().getNumber());
+                where.appendRequired("acct.phone.regionCode=:regionCode", "regionCode", unguarded.getPhone().getRegionCode());
             } else if (unguarded.getSynapseUserId() != null) {
                 where.append("acct.synapseUserId=:synapseUserId", "synapseUserId", unguarded.getSynapseUserId());
             } else {
                 where.append("enrollment.externalId=:externalId", "externalId", unguarded.getExternalId());
             }
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java:164]: All of these should be appendRequired

File: src/main/java/org/sagebionetworks/bridge/services/OrganizationService.java [hunk 2]
```diff
@@ -210,8 +209,7 @@ public void deleteOrganization(String appId, String identifier) {
         checkArgument(isNotBlank(appId));
         checkNotNull(search);
 
-        AccountSummarySearch scopedSearch = new AccountSummarySearch.Builder()
-            .copyOf(search)
+        AccountSummarySearch scopedSearch = search.toBuilder()
             .withAdminOnly(true)
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/services/OrganizationService.java:213]: Why is this one true when the one above is null?

File: src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java [hunk 3]
```diff
@@ -221,8 +222,7 @@ public Timeline getTimelineForUser(@PathVariable String studyId, @PathVariable S
         App app = appService.getApp(session.getAppId());
         AccountSummarySearch search = parseJson(AccountSummarySearch.class);
         
-        search = new AccountSummarySearch.Builder().copyOf(search)
-                .withEnrolledInStudyId(studyId).build();
+        search = search.toBuilder().withEnrolledInStudyId(studyId).withAdminOnly(FALSE).build();
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java]: Revert this.

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java [hunk 4]
```diff
@@ -372,7 +374,8 @@ public void searchForAccountSummaries() throws Exception {
         when(mockParticipantService.getPagedAccountSummaries(eq(app), any())).thenReturn(page);
         
         AccountSummarySearch search = new AccountSummarySearch.Builder()
-                .withAdminOnly(true).build();
+                .withAdminOnly(TRUE)
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java:377]: remove this? we only care to return enrolled, whether admins or not

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java:164]: Changed (no difference in the SQL output, but clearer).

File: src/main/java/org/sagebionetworks/bridge/services/OrganizationService.java (hunk 2 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/OrganizationService.java:213]: This is fine, the method is called "getUnassignedAdmins"

File: src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java (hunk 3 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java]: removed

**Review by nategbrown9 — APPROVED** [review]
Looks good.

**Pull request merged**